### PR TITLE
remove `mock-property` types, now that it bundles types itself

### PR DESCRIPTION
**Pull request by ljharb**
Added in #67012

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:

- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by ljharb** [comment]
@jakebailey this failure seems similar to #68945?

File: notNeededPackages.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
             "libraryName": "hyphenate-style-name",
             "asOfVersion": "1.1.0"
         },
+        "@types/mock-property": {
+            "libraryName": "@types/mock-property",
```
Comment (jakebailey) [inline notNeededPackages.json:12]:
```suggestion
        "mock-property": {
            "libraryName": "mock-property",
```

This is probably why things aren't working.

Comment (jakebailey) [reply notNeededPackages.json]: (It's on my list to try and fix the script to disallow this, there's at least one of these oopses checked in already)

**Review by jakebailey — APPROVED** [review]

**Comment by ljharb** [comment]
Ready to merge

**Pull request merged**